compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -2996,8 +3012,15 @@ private ParserRuleContext getNextRuleInternal(ParserRuleContext currentCtx, int
                 return ParserRuleContext.LT;
             case LT:
                 return getNextRuleForLt();
+            case INFERRED_TYPEDESC_DEFAULT_START_LT:
+                return ParserRuleContext.INFERRED_TYPEDESC_DEFAULT_END_GT;
             case GT:
                 return getNextRuleForGt(nextLookahead);
+            case INFERRED_TYPEDESC_DEFAULT_END_GT:
+                return getNextRuleForExpr();
```

Comment:
Since after `<>` the default value should end we should not call `getNextRuleForExpr()`. Use an alternative rule array with  `{ ParserRuleContext.CLOSE_PARENTHESIS, ParserRuleContext.COMMA }`